### PR TITLE
Additional Pipeline Compatibility

### PR DESCRIPTION
**Pull request by robert-shade**
When using a pipeline definition similar to the following, the server could not be found.  This PR fixes deserializing AccurevSCM when only a serverUUID is given.

Since Jenkins 2.60 is targeted in the pom.xml, this also removes the usage of several deprecated methods.

```
pipeline {
    agent {
        node {
            label "test"
        }
    }
    stages {
        stage('Checkout Source') {
            steps {
                checkout([$class: 'AccurevSCM',
                          depot: 'Test_Source',
                          serverUUID: 'XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX',
                          stream: 'TestStr',
                          wspaceORreftree: 'none',
                          subPath: 'sw/utl/scripts'])
            }
        }
    }
}
```

### CODE REVIEW

File: src/main/java/jenkins/plugins/accurev/util/AccurevUtils.java [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 import static org.apache.commons.lang.time.DateUtils.MILLIS_PER_SECOND;
 
-import edu.umd.cs.findbugs.annotations.CheckForNull;
 import hudson.FilePath;
 import hudson.model.Computer;
 import hudson.model.Node;
 import java.util.Date;
+import javax.annotation.CheckForNull;
```
Comment (casz) [inline src/main/java/jenkins/plugins/accurev/util/AccurevUtils.java]: No no no... License issue and the whole JSR 305 is not in Java 8, I don't know the solution on then to keep using FindBugs annotation.

File: src/main/java/hudson/plugins/accurev/AccurevSCM.java [hunk 2]
```diff
@@ -124,6 +125,19 @@ public AccurevSCM(String serverName, String depot, String stream) {
     updateMode();
   }
 
+  @DataBoundConstructor
+  public AccurevSCM(String serverName, String serverUUID, String depot, String stream) {
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccurevSCM.java]: Why is this change needed?

Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccurevSCM.java]: Why is this change needed? if so should the deprecated method not call this one.

**Comment by casz** [comment]
Perhaps we should have a symbol to avoid `$class: 'AccurevSCM'`

I just checked... For some reason I added a symbol `accurev` to `AccurevTool` which seems wrong. Perhaps best renamed to `accurevTool` and then for accurevSCM we register the symbol `accurev` 😇

**Comment by casz** [comment]
Seems findbugs is complaining about serialization.

File: src/main/java/jenkins/plugins/accurev/util/AccurevUtils.java (hunk 1 above)
Comment (robert-shade) [reply src/main/java/jenkins/plugins/accurev/util/AccurevUtils.java]: Roger

File: src/main/java/hudson/plugins/accurev/AccurevSCM.java (hunk 2 above)
Comment (robert-shade) [reply src/main/java/hudson/plugins/accurev/AccurevSCM.java]: One of the tests needed this constructor - i'll simplify the deprecated one to call this

File: src/main/java/hudson/plugins/accurev/AccurevSCM.java [hunk 3]
```diff
@@ -991,6 +995,10 @@ public DescriptorImpl getDescriptor() {
     @Extension
     public static class DescriptorImpl extends Descriptor<AccurevServer> {
 
+      public DescriptorImpl() {
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccurevSCM.java]: 🙌 Would be nice if you could actually add a unit test for JCasc cause this change enables accurev servers to be managed from JCasc.
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/PLUGINS.md#how-to-test-

File: src/main/java/hudson/plugins/accurev/AccurevSCM.java [hunk 4]
```diff
@@ -991,6 +995,10 @@ public DescriptorImpl getDescriptor() {
     @Extension
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccurevSCM.java]: Add a accurevServers symbol for jcasc 🤭

File: pom.xml [hunk 5]
```diff
@@ -45,6 +45,12 @@
       <version>2.8.9</version>
       <scope>test</scope>
     </dependency>
+	<dependency>
+      <groupId>io.jenkins</groupId>
+      <artifactId>configuration-as-code</artifactId>
+      <version>1.0</version>
```
Comment (casz) [inline pom.xml]: Please bump to 1.2 😇

File: src/main/java/hudson/plugins/accurev/AccuRevHiddenParametersAction.java [hunk 6]
```diff
@@ -14,7 +14,10 @@ public AccuRevHiddenParametersAction(EnvVars values) {
     this.values = values;
   }
 
-  public void buildEnvVars(AbstractBuild<?, ?> build, EnvVars env) {
+  @Override
+  public void buildEnvironment(
+      @SuppressWarnings("deprecation") @edu.umd.cs.findbugs.annotations.NonNull Run<?, ?> run,
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccuRevHiddenParametersAction.java]: I think it would better to find a solution rather then to pile on the suppression of deprecation.
Either choose to ignore them or find a solution in
Perhaps JetBrains annotation:
https://mvnrepository.com/artifact/org.jetbrains/annotations/16.0.3
They are under Apache 2 license so should be fine.
Or Eclipse annotation:
https://mvnrepository.com/artifact/org.eclipse.jdt/org.eclipse.jdt.annotation/2.2.100

Suppressing is definitely not the solution 😅

Comment (casz) [reply src/main/java/hudson/plugins/accurev/AccuRevHiddenParametersAction.java]: Perhaps spotbugs its a better option?

Comment (robert-shade) [reply src/main/java/hudson/plugins/accurev/AccuRevHiddenParametersAction.java]: Looking at Jenkins itself, it looks like they use the javax.annotation annotations (https://github.com/jenkinsci/jenkins/search?q=NonNull&unscoped_q=NonNull).  Any reason why we wouldn't do that, too?

Comment (casz) [reply src/main/java/hudson/plugins/accurev/AccuRevHiddenParametersAction.java]: I am inclined to agree with @stephenc (sorry for tagging 😇) https://blog.codefx.org/java/jsr-305-java-9/#comment-3582702088

File: pom.xml [hunk 7]
```diff
@@ -48,9 +48,14 @@
 	<dependency>
       <groupId>io.jenkins</groupId>
       <artifactId>configuration-as-code</artifactId>
-      <version>1.0</version>
+      <version>1.2</version>
       <scope>test</scope>
 	</dependency>
+	<dependency>
+		<groupId>org.eclipse.jdt</groupId>
+		<artifactId>org.eclipse.jdt.annotation</artifactId>
```
Comment (stephenc) [inline pom.xml]: The spotbugs ones would be better imho. Can we all just standardise on the spotbugs ones (and bonus they are the same package as the original findbugs ones)

Comment (casz) [reply pom.xml]: I agree 🙌 Plus I started https://github.com/jenkinsci/plugin-pom/pull/128

Comment (robert-shade) [reply pom.xml]: Any thoughts on where we should go with this now?  Looks like the mentioned PR is hitting some roadblocks.

Comment (casz) [reply pom.xml]: We can switch to spotbugs in this repo 😅 and later remove it once plugin pom is ready...

**Comment by robert-shade** [comment]
@casz anything else you're looking for before merging?

File: src/main/java/hudson/plugins/accurev/AccuRevHiddenParametersAction.java (hunk 6 above)
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccuRevHiddenParametersAction.java]: Please remove `@SuppressWarnings("deprecation")`

File: src/main/java/hudson/plugins/accurev/AccuRevHiddenParametersAction.java [hunk 8]
```diff
@@ -14,7 +14,10 @@ public AccuRevHiddenParametersAction(EnvVars values) {
...
+      @SuppressWarnings("deprecation") @edu.umd.cs.findbugs.annotations.NonNull EnvVars env) {
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccuRevHiddenParametersAction.java]: Please remove `@SuppressWarnings("deprecation")`
